service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java

```diff
@@ -40,26 +40,23 @@
     public abstract Optional<SslConfiguration> security();
 
     /**
+     * Connect timeout for requests in ms.
+     */
+    public abstract Optional<Integer> connectTimeoutMs();
```

Comment:
I guess I was thinking of the Dropwizard Duration format, which might be useful to extract to a more general library for things like this, which allows encoding things like "100 milliseconds" and "100 ms" as well as "10 seconds" or "2d" (for 2 days).